RadixWallet/Features/AccountRecoveryScan/Children/AccountRecoveryScanInProgress/AccountRecoveryScanInProgress.swift

```diff
@@ -112,49 +136,75 @@ public struct AccountRecoveryScanInProgress: Sendable, FeatureReducer {
+				return .run { [networkID = state.networkID] send in
+					let result = await TaskResult<IndicesUsedByFactorSource> {
+						try await factorSourcesClient.indicesOfEntitiesControlledByFactorSource(
+							.init(
+								entityKind: .account,
+								factorSourceID: id.embed(),
+								networkID: networkID
+							)
+						)
 					}
-					await send(.internal(.loadFactorSourceResult(result)))
+					await send(.internal(.loadIndicesUsedByFactorSourceResult(result)))
 				}
 			}
 
 		case .scanMore:
-			guard let factorSource = state.factorSource.wrappedValue else { fatalError("discrepancy") }
-			state.offset += accRecScanBatchSize
-			return derivePublicKeys(using: factorSource, state: &state)
+			loggerGlobal.debug("Scan more requested.")
+			return derivePublicKeys(state: &state)
 
 		case .continueTapped:
-			return .send(.delegate(.foundAccounts(active: state.active, inactive: state.inactive)))
+			if let maxActive = state.active.max() {
+				let inactiveInBetweenActive = state.inactive.filter {
+					$0.derivationIndex < maxActive.derivationIndex
+				}
+				return .send(.delegate(.foundAccounts(active: state.active, inactive: inactiveInBetweenActive)))
+			} else {
+				return .send(.delegate(.foundAccounts(active: [], inactive: [])))
+			}
 		}
 	}
 
-	public func reduce(into state: inout State, childAction: ChildAction) -> Effect<Action> {
-		switch childAction {
-		/// Attention! If you change this to the `destination` pattern we use elsewhere this feature breaks due to "TCA Send"-bug
-		case let .derivePublicKeys(.presented(.delegate(delegateAction))):
-			loggerGlobal.notice("Finish deriving public keys")
+	public func reduce(into state: inout State, presentedAction: Destination.Action) -> Effect<Action> {
+		switch presentedAction {
+		case let .derivePublicKeys(.delegate(delegateAction)):
+			loggerGlobal.notice("Finish deriving public keys => `state.destination = nil`")
 			switch delegateAction {
 			case let .derivedPublicKeys(publicHDKeys, factorSourceID, networkID):
-				assert(factorSourceID == state.factorSourceID.embed())
+				let id = state.factorSourceIDFromHash
+				assert(factorSourceID == id.embed())
 				assert(networkID == state.networkID)
-				return .run { [id = state.factorSourceID] send in
-					let accounts = await publicHDKeys.enumerated().asyncMap { offset, publicHDKey in
+				return .run { send in
+					let accounts = await publicHDKeys.enumerated().asyncMap {
+						offset,
+							publicHDKey in
 						let appearanceID = await accountsClient.nextAppearanceID(networkID, offset)
 						let account = try! Profile.Network.Account(
 							networkID: networkID,
 							factorInstance: .init(
 								factorSourceID: id,
 								publicHDKey: publicHDKey
 							),
-							displayName: "Unnamed",
-							extraProperties: .init(appearanceID: appearanceID)
+							displayName: "Unnamed", // FIXME: Strings
+							extraProperties: .init(
+								appearanceID: appearanceID,
+								onLedgerSettings: .unknown
```

Comment:
we update `onLedgerSettings.depositRule` later in this reducer, when we do the scanning of the network!